specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2089 @@
+        "isArchiveEnabled": {
+          "description": "The value that indicates whether archiving is enabled or not.",
+          "type": "boolean"
+        }
+      }
+    },
+    "BuildGetLogResult": {
+      "description": "The result of get log link operation.",
+      "type": "object",
+      "properties": {
+        "logLink": {
+          "description": "The link to logs for a azure container registry build.",
+          "type": "string"
+        }
+      }
+    },
+    "BuildStepList": {
+      "description": "The collection of build items.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The collection value.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildStep"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next set of paged results.",
+          "type": "string"
+        }
+      }
+    },
+    "BuildStep": {
+      "description": "Build step resource properties",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildStepProperties",
+          "description": "The properties of a build step.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "BuildStepProperties": {
+      "description": "Base properties for any  build step.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "The provisioning state of the build step.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "type": {
+          "description": "The type of the step.",
+          "enum": [
+            "Docker"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "BuildStepType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "type"
+    },
+    "BuildStepUpdateParameters": {
+      "description": "The parameters for updating a build step.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildStepPropertiesUpdateParameters",
+          "description": "The properties for updating a build step.",
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The ARM resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "BuildStepPropertiesUpdateParameters": {
```

Comment:
>BuildStepPropertiesUpdateParameters [](start = 5, length = 35)

Since `BuildStepProperties` has a discriminator, should the update stuff also have it?

Reply:
It is, but I did not annotate with knowntype. Thanks Will do.

Reply:
Fixed